pkg/rancher-desktop/backend/backendHelper.ts

```diff
@@ -220,7 +222,8 @@ export default class BackendHelper {
         throw new LockedFieldError(`Locked kubernetes version '${ currentConfigVersionString }' isn't a valid version.`);
       }
       const message = invalidK8sVersionMainMessage;
-      const detail = `Falling back to the most recent stable version of ${ availableVersions[0] }`;
+      // firstStableVersion is guaranteed to return a version because availableVersions is not empty.
```

Comment:
While true, it's easier to not use `!` in TypeScript (mostly because I think some linter would complain about it, but I guess it no longer does?)

```suggestion
      const stableVersion = firstStableVersion(availableVersions);

      if (!stableVersion) {
         // This should never be reached, as `availableVersions` should be non-empty.
         throw new Error(`Failed to find stable version`);
…
```

Reply:
I disagree about it being "easier" to not use `!`, but I made the change because the relationship between `availableVersions` being non-empty and `firstStableVersion` not being undefined isn't explicit.

But to avoid code duplication I had to move the setting of `stableVersion` way up in the function, out of the current scope, because the same expression is used further down once more.

So everything is spread out more, `firstStableVersion` is called when we don't need the result, and it is now a `K8s.VersionEntry` instead of a `semver.SemVer`, so must be used as `stableEntry.version.version`.

Also TypeScript doesn't give a warning when you forget to add the second `.version`, but that isn't really different from before.

Given that there haven't been problems with using `!` and the linter, can we stop avoiding it? I also wanted to change:

```diff
- new VersionEntry(semver.parse('1.99.3+k3s1') as semver.SemVer, ['stable']),
+ new VersionEntry(semver.parse('1.99.3+k3s1')!, ['stable']),
```

but have undone those changes for now.